rtengine/imageio.cc

```diff
@@ -83,6 +85,43 @@ Glib::ustring to_utf8 (const std::string& str)
     }
 }
 
+unsigned short addSat (unsigned short a, unsigned short b)
+{
+    unsigned short res = a + b;
+    res |= -(res < a);
+    return res;
+}
+
+void scanlineToPseudoGrey (const unsigned short *in16, unsigned char *out8, unsigned long width)
+{
+    // For details see http://r0k.us/graphics/PseudoGreyPlus.html
+
+    static const unsigned short plusses[16][3] = {
+        {0 << 8, 0 << 8, 0 << 8},
+        {0 << 8, 0 << 8, 1 << 8},
+        {0 << 8, 0 << 8, 2 << 8},
+        {1 << 8, 0 << 8, 0 << 8},
+        {1 << 8, 0 << 8, 1 << 8},
+        {1 << 8, 0 << 8, 1 << 8},
+        {1 << 8, 0 << 8, 2 << 8},
+        {2 << 8, 0 << 8, 0 << 8},
+        {2 << 8, 0 << 8, 1 << 8},
+        {2 << 8, 0 << 8, 2 << 8},
+        {2 << 8, 0 << 8, 2 << 8},
+        {0 << 8, 1 << 8, 0 << 8},
+        {0 << 8, 1 << 8, 1 << 8},
+        {0 << 8, 1 << 8, 2 << 8},
+        {0 << 8, 1 << 8, 2 << 8},
+        {1 << 8, 1 << 8, 0 << 8}
+    };
+
+    for (unsigned long x = 0; x < width * 3; x += 3) {
```

Comment:
Could probably benefit from using pointer arithmetic instead of indexing? But maybe the compiler will do that itself...